include/_mgp_mock.py

```diff
@@ -0,0 +1,453 @@
+
+    def __init__(self, id: int, graph: Graph) -> None:
+        if not isinstance(id, int):
+            raise TypeError(f"Expected 'int', got '{type(id)}'")
+
+        if not isinstance(graph, Graph):
+            raise TypeError(f"Expected '_mgp_mock.Graph', got '{type(graph)}'")
+
+        if not graph.nx.has_node(id):
+            raise IndexError(f"Unable to find vertex with ID {id}.")
+
+        self._id = id
+        self._graph = graph
+
+    def is_valid(self) -> bool:
+        return self._graph.is_valid()
+
+    def is_deleted(self) -> bool:
+        return not self._graph.nx.has_node(self._id) and self._id <= self._graph.highest_vertex_id
+
+    @property
+    def underlying_graph(self) -> Graph:
+        return self._graph
+
+    def underlying_graph_is_mutable(self) -> bool:
+        return not nx.is_frozen(self._graph.nx)
+
+    @property
+    def labels(self) -> typing.List[int]:
+        return self._graph.nx.nodes[self._id][NX_LABEL_ATTR].split(":")
+
+    def add_label(self, label: str) -> None:
+        if nx.is_frozen(self._graph.nx):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        self._graph.nx.nodes[self._id][NX_LABEL_ATTR] += f":{label}"
+
+    def remove_label(self, label: str) -> None:
+        if nx.is_frozen(self._graph.nx):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        labels = self._graph.nx.nodes[self._id][NX_LABEL_ATTR]
+        if labels.startswith(f"{label}:"):
+            labels = "\n" + labels  # pseudo-string starter
+            self._graph.nx.nodes[self._id][NX_LABEL_ATTR] = labels.replace(f"\n{label}:", "")
+        elif labels.endswith(f":{label}"):
+            labels += "\n"  # pseudo-string terminator
+            self._graph.nx.nodes[self._id][NX_LABEL_ATTR] = labels.replace(f":{label}\n", "")
+        else:
+            self._graph.nx.nodes[self._id][NX_LABEL_ATTR] = labels.replace(f":{label}:", ":")
+
+    @property
+    def id(self) -> int:
+        return self._id
+
+    @property
+    def properties(self):
+        return (
+            (key, value)
+            for key, value in self._graph.nx.nodes[self._id].items()
+            if key not in (NX_LABEL_ATTR, NX_TYPE_ATTR)
+        )
+
+    def get_property(self, property_name: str):
+        return self._graph.nx.nodes[self._id][property_name]
+
+    def set_property(self, property_name: str, value: object):
+        self._graph.nx.nodes[self._id][property_name] = value
+
+    @property
+    def in_edges(self) -> typing.Iterable["Edge"]:
+        return [Edge(edge, self._graph) for edge in self._graph.nx.in_edges(self._id, keys=True)]
+
+    @property
+    def out_edges(self) -> typing.Iterable["Edge"]:
+        return [Edge(edge, self._graph) for edge in self._graph.nx.out_edges(self._id, keys=True)]
+
+
+class Edge:
+    """Represents a graph edge."""
+
+    __slots__ = ("_edge", "_graph")
+
+    I_START = 0
+    I_END = 1
+    I_KEY = 2
```

Comment:
This you could isolate inside `class Constants`

Reply:
Done